rust/private/rust.bzl

```diff
@@ -124,19 +152,26 @@ def _rust_test_common(ctx, test_binary):
     """
     toolchain = find_toolchain(ctx)
 
-    if len(ctx.attr.deps) == 1 and len(ctx.files.srcs) == 0:
-        # Target has a single dependency but no srcs. Build the test binary using
-        # the dependency's srcs.
-        parent_crate = ctx.attr.deps[0][CrateInfo]
+    if ctx.attr.crate:
+        # Target is building the crate in `test` config
+        # Build the test binary using the dependency's srcs.
+        crate = ctx.attr.crate[CrateInfo]
         target = CrateInfo(
             name = test_binary.basename,
-            type = parent_crate.type,
-            root = parent_crate.root,
-            srcs = parent_crate.srcs,
-            deps = parent_crate.deps,
+            type = crate.type,
+            root = crate.root,
+            srcs = crate.srcs + ctx.files.srcs,
```

Comment:
The use-case for this is additional test-only files?

Reply:
yes